openml-lightgbm/lightgbm-provider/src/main/java/com/feedzai/openml/provider/lightgbm/LightGBMSWIG.java

```diff
@@ -132,71 +121,38 @@ private void initSWIGInstanceArray(final Instance instance) {
      * @param instance The instance from pulse.
      * @return double[2] array with scores.
      */
-    double[] getBinaryClassDistribution(final Instance instance) {
+    public double[] getBinaryClassDistribution(final Instance instance) {
 
         // we need to lock the resources to avoid having multiple threads sharing the same swig resources.
         synchronized (this.swigResources) {
 
-            initSWIGInstanceArray(instance);
-
-            // LightGBM call configuration:
-            final int isRowMajor = 1;
-            final int numIterations = -1;
-            final String LightGBMParameters = "num_threads=1";
-
-            final int returnCodeLGBM = lightgbmlibJNI.LGBM_BoosterPredictForMatSingleRow(
-                    this.swigResources.swigBoosterHandle,
+            copyDataToSWIGInstance(instance);
+            final int returnCodeLGBM = lightgbmlibJNI.LGBM_BoosterPredictForMatSingleRowFast(
+                    this.swigResources.swigFastConfigHandle,
                     this.swigResources.swigInstancePtr,
-                    lightgbmlibConstants.C_API_DTYPE_FLOAT64,
-                    this.schemaNumFeatures,
-                    isRowMajor,
-                    lightgbmlibConstants.C_API_PREDICT_NORMAL,
-                    0, // start at iteration 0
-                    numIterations,
-                    LightGBMParameters,
-                    // useless API output: size known already (had to preallocate memory)
                     this.swigResources.swigOutLengthInt64Ptr,
                     this.swigResources.swigOutScoresPtr // preallocated memory
             );
 
             if (returnCodeLGBM == -1)
                 throw new LightGBMException();
 
+            // Valid for binary predictions:
```

Comment:
The comment isn't clear
is it predicting a valid binary score?

Reply:
True, will remove :p